### PR TITLE
DM XMLs 1.4 tag

### PR DESCRIPTION
**Pull request by cecille**
Changes from 1.3:
====================
Clusters newly added in 1.4
['Joint Fabric PKI', 'Water Heater Mode', 'Water Heater Management', 'Thread Network Directory', 'Ecosystem Information', 'Water Tank Level Monitoring', 'Thread Border Router Management', 'Service Area', 'Commissioner Control', 'Joint Fabric Datastore Cluster', 'Wi-Fi Network Management']

Clusters removed since 1.3
['Pulse Width Modulation']

Cluster Thermostat
	Revision change - old: 6 new: 8
	Features added: ['MatterScheduleConfiguration', 'Presets']
	Attributes added: ['PresetTypes', 'ScheduleTypes', 'NumberOfPresets', 'NumberOfSchedules', 'NumberOfScheduleTransitions', 'NumberOfScheduleTransitionPerDay', 'ActivePresetHandle', 'ActiveScheduleHandle', 'Presets', 'Schedules', 'SetpointHoldExpiryTimestamp']
	Attributes changed:
		Occupancy
			Occupancy: datatype: ref_OccupancyBitmap conformance: OCC, access = R V
			Occupancy: datatype: OccupancyBitmap conformance: OCC, access = R V
	Accepted Commands added: ['SetActiveScheduleRequest', 'SetActivePresetRequest', 'Atomic Request']
	Generated Commands added: ['Atomic Response']

Cluster Identify
	Revision change - old: 4 new: 5

Cluster Occupancy Sensing
	Revision change - old: 4 new: 5
	Features added: ['Other', 'PassiveInfrared', 'Ultrasonic', 'PhysicalContact', 'ActiveInfrared', 'Radar', 'RFSensing', 'Vision']
	Attributes added: ['HoldTime', 'HoldTimeLimits']
	Attributes changed:
		OccupancySensorType
			OccupancySensorType: datatype: OccupancySensorTypeEnum conformance: M, access = R V
			OccupancySensorType: datatype: OccupancySensorTypeEnum conformance: M, D, access = R V
		OccupancySensorTypeBitmap
			OccupancySensorTypeBitmap: datatype: OccupancySensorTypeBitmap conformance: M, access = R V
			OccupancySensorTypeBitmap: datatype: OccupancySensorTypeBitmap conformance: M, D, access = R V
		UltrasonicOccupiedToUnoccupiedDelay
			UltrasonicOccupiedToUnoccupiedDelay: datatype: uint16 conformance: O, access = RW VM
			UltrasonicOccupiedToUnoccupiedDelay: datatype: uint16 conformance: [HoldTime & US], D, access = RW VM
		UltrasonicUnoccupiedToOccupiedDelay
			UltrasonicUnoccupiedToOccupiedDelay: datatype: uint16 conformance: UltrasonicUnoccupiedToOccupiedThreshold, O, access = RW VM
			UltrasonicUnoccupiedToOccupiedDelay: datatype: uint16 conformance: HoldTime & US & UltrasonicUnoccupiedToOccupiedThreshold, [HoldTime & US], D, access = RW VM
		UltrasonicUnoccupiedToOccupiedThreshold
			UltrasonicUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: UltrasonicUnoccupiedToOccupiedDelay, O, access = RW VM
			UltrasonicUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: HoldTime & US & UltrasonicUnoccupiedToOccupiedDelay, [HoldTime & US], D, access = RW VM
		PIROccupiedToUnoccupiedDelay
			PIROccupiedToUnoccupiedDelay: datatype: uint16 conformance: O, access = RW VM
			PIROccupiedToUnoccupiedDelay: datatype: uint16 conformance: [HoldTime & (PIR | (!PIR & !US & !PHY))], D, access = RW VM
		PIRUnoccupiedToOccupiedDelay
			PIRUnoccupiedToOccupiedDelay: datatype: uint16 conformance: PIRUnoccupiedToOccupiedThreshold, O, access = RW VM
			PIRUnoccupiedToOccupiedDelay: datatype: uint16 conformance: (HoldTime & (PIR | (!PIR & !US & !PHY))) & PIRUnoccupiedToOccupiedThreshold, [HoldTime & (PIR | (!PIR & !US & !PHY))], D, access = RW VM
		PIRUnoccupiedToOccupiedThreshold
			PIRUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: PIRUnoccupiedToOccupiedDelay, O, access = RW VM
			PIRUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: (HoldTime & (PIR | (!PIR & !US & !PHY))) & PIRUnoccupiedToOccupiedDelay, [HoldTime & (PIR | (!PIR & !US & !PHY))], D, access = RW VM
		PhysicalContactOccupiedToUnoccupiedDelay
			PhysicalContactOccupiedToUnoccupiedDelay: datatype: uint16 conformance: O, access = RW VM
			PhysicalContactOccupiedToUnoccupiedDelay: datatype: uint16 conformance: [HoldTime & PHY], D, access = RW VM
		PhysicalContactUnoccupiedToOccupiedDelay
			PhysicalContactUnoccupiedToOccupiedDelay: datatype: uint16 conformance: O, access = RW VM
			PhysicalContactUnoccupiedToOccupiedDelay: datatype: uint16 conformance: HoldTime & PHY & PhysicalContactUnoccupiedToOccupiedThreshold, [HoldTime & PHY], D, access = RW VM
		PhysicalContactUnoccupiedToOccupiedThreshold
			PhysicalContactUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: PhysicalContactUnoccupiedToOccupiedDelay, O, access = RW VM
			PhysicalContactUnoccupiedToOccupiedThreshold: datatype: uint8 conformance: HoldTime & PHY & PhysicalContactUnoccupiedToOccupiedDelay, [HoldTime & PHY], D, access = RW VM
	Events added: ['OccupancyChanged']

Cluster Level Control
	Revision change - old: 5 new: 6

Cluster Pump Configuration and Control
	Attributes changed:
		AlarmMask
			AlarmMask: datatype: map16 conformance: D, access = R V
			AlarmMask: datatype: UNKNOWN conformance: D, access = RW N/AN/A

Cluster AccessControl
	Revision change - old: 1 new: 2
	Features added: ['Extension', 'ManagedDevice']
	Attributes added: ['CommissioningARL', 'ARL']
	Attributes changed:
		Extension
			Extension: datatype: list conformance: O, access = RW AA
			Extension: datatype: list conformance: EXTS, access = RW AA
	Accepted Commands added: ['ReviewFabricRestrictions']
	Generated Commands added: ['ReviewFabricRestrictionsResponse']
	Events added: ['FabricRestrictionReviewUpdate']
	Events changed:
		AccessControlExtensionChanged
			XmlEvent(name='AccessControlExtensionChanged', conformance=mandatory(choice=None))
			XmlEvent(name='AccessControlExtensionChanged', conformance=mandatory_wrapper(choice=None))

Cluster Basic Information
	Revision change - old: 3 new: 4
	Attributes changed:
		UniqueID
			UniqueID: datatype: string conformance: O, access = R V
			UniqueID: datatype: string conformance: M, access = R V

Cluster Power Source
	Revision change - old: 2 new: 3
	Features changed:
		Wired
			XmlFeature(code='WIRED', name='Wired', conformance=optional(choice=None))
			XmlFeature(code='WIRED', name='Wired', conformance=optional(choice=Choice(marker='a', more=False)))
		Battery
			XmlFeature(code='BAT', name='Battery', conformance=optional(choice=None))
			XmlFeature(code='BAT', name='Battery', conformance=optional(choice=Choice(marker='a', more=False)))

Cluster General Commissioning
	Revision change - old: 1 new: 2
	Features added: ['TermsAndConditions']
	Attributes added: ['TCAcceptedVersion', 'TCMinRequiredVersion', 'TCAcknowledgements', 'TCAcknowledgementsRequired', 'TCUpdateDeadline']
	Accepted Commands added: ['SetTCAcknowledgements']
	Generated Commands added: ['SetTCAcknowledgementsResponse']

Cluster Thread Network Diagnostics
	Revision change - old: 2 new: 3
	Attributes added: ['ExtAddress', 'Rloc16']

Cluster Bridged Device Basic Information
	Revision change - old: 3 new: 4
	Features added: ['BridgedICDSupport']
	Attributes changed:
		ProductID
			ProductID: datatype: uint16 conformance: X, access = R V
			ProductID: datatype: uint16 conformance: O, access = R V
		UniqueID
			UniqueID: datatype: string conformance: O, access = R V
			UniqueID: datatype: string conformance: M, access = R V
	Accepted Commands added: ['KeepActive']
	Events added: ['ActiveChanged']

Cluster Switch
	Revision change - old: 1 new: 2
	Features added: ['ActionSwitch']
	Features changed:
		MomentarySwitchRelease
			XmlFeature(code='MSR', name='MomentarySwitchRelease', conformance=optional_wrapper(choice=None))
			XmlFeature(code='MSR', name='MomentarySwitchRelease', conformance=optional_wrapper(choice=None))
		MomentarySwitchLongPress
			XmlFeature(code='MSL', name='MomentarySwitchLongPress', conformance=optional_wrapper(choice=None))
			XmlFeature(code='MSL', name='MomentarySwitchLongPress', conformance=optional_wrapper(choice=None))
		MomentarySwitchMultiPress
			XmlFeature(code='MSM', name='MomentarySwitchMultiPress', conformance=optional_wrapper(choice=None))
			XmlFeature(code='MSM', name='MomentarySwitchMultiPress', conformance=otherwise(choice=None))
	Events changed:
		MultiPressOngoing
			XmlEvent(name='MultiPressOngoing', conformance=mandatory_wrapper(choice=None))
			XmlEvent(name='MultiPressOngoing', conformance=mandatory_wrapper(choice=None))

Cluster GroupKeyManagement
	Features changed:
		CacheAndSync
			XmlFeature(code='CS', name='CacheAndSync', conformance=optional(choice=None))
			XmlFeature(code='CS', name='CacheAndSync', conformance=provisional(choice=None))

Cluster ICDManagement
	Revision change - old: 2 new: 3
	Features added: ['DynamicSitLitSupport']
	Features changed:
		CheckInProtocolSupport
			XmlFeature(code='CIP', name='CheckInProtocolSupport', conformance=otherwise(choice=None))
			XmlFeature(code='CIP', name='CheckInProtocolSupport', conformance=otherwise(choice=None))
		UserActiveModeTrigger
			XmlFeature(code='UAT', name='UserActiveModeTrigger', conformance=otherwise(choice=None))
			XmlFeature(code='UAT', name='UserActiveModeTrigger', conformance=otherwise(choice=None))
		LongIdleTimeSupport
			XmlFeature(code='LITS', name='LongIdleTimeSupport', conformance=otherwise(choice=None))
			XmlFeature(code='LITS', name='LongIdleTimeSupport', conformance=optional(choice=None))
	Attributes added: ['MaximumCheckInBackoff']
	Attributes changed:
		UserActiveModeTriggerHint
			UserActiveModeTriggerHint: datatype: UserActiveModeTriggerBitmap conformance: P, UAT, access = R V
			UserActiveModeTriggerHint: datatype: UserActiveModeTriggerBitmap conformance: UAT, access = R V
		OperatingMode
			OperatingMode: datatype: OperatingModeEnum conformance: P, LITS, access = R V
			OperatingMode: datatype: OperatingModeEnum conformance: LITS, access = R V

Cluster Oven Cavity Operational State
	Revision change - old: 1 new: 2
	Accepted Commands changed:
		Pause
			XmlCommand(id=0, name='Pause', conformance=otherwise(choice=None))
			XmlCommand(id=0, name='Pause', conformance=disallowed(choice=None))
		Stop
			XmlCommand(id=1, name='Stop', conformance=otherwise(choice=None))
			XmlCommand(id=1, name='Stop', conformance=optional(choice=None))
		Resume
			XmlCommand(id=3, name='Resume', conformance=otherwise(choice=None))
			XmlCommand(id=3, name='Resume', conformance=disallowed(choice=None))
	Generated Commands changed:
		OperationalCommandResponse
			XmlCommand(id=4, name='OperationalCommandResponse', conformance=mandatory_wrapper(choice=None))
			XmlCommand(id=4, name='OperationalCommandResponse', conformance=optional(choice=None))
	Events changed:
		OperationCompletion
			XmlEvent(name='OperationCompletion', conformance=otherwise(choice=None))
			XmlEvent(name='OperationCompletion', conformance=optional(choice=None))

Cluster Oven Mode
	Revision change - old: 1 new: 2
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: O, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: DEPONOFF, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Laundry Washer Mode
	Revision change - old: 2 new: 3
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: P, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: P, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Refrigerator And Temperature Controlled Cabinet Mode
	Revision change - old: 2 new: 3
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: P, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: P, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Laundry Washer Controls
	Revision change - old: 1 new: 2
	Features changed:
		Spin
			XmlFeature(code='SPIN', name='Spin', conformance=optional(choice=None))
			XmlFeature(code='SPIN', name='Spin', conformance=optional(choice=Choice(marker='a', more=True)))
		Rinse
			XmlFeature(code='RINSE', name='Rinse', conformance=optional(choice=None))
			XmlFeature(code='RINSE', name='Rinse', conformance=optional(choice=Choice(marker='a', more=True)))

Cluster RVC Run Mode
	Revision change - old: 2 new: 3
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		OnMode
			OnMode: datatype: uint8 conformance: D, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster RVC Clean Mode
	Revision change - old: 2 new: 3
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		OnMode
			OnMode: datatype: uint8 conformance: D, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Dishwasher Mode
	Revision change - old: 2 new: 3
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: P, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: P, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Microwave Oven Mode
	Revision change - old: 1 new: 2
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))

Cluster Operational State
	Revision change - old: 2 new: 3
	Events changed:
		OperationCompletion
			XmlEvent(name='OperationCompletion', conformance=otherwise(choice=None))
			XmlEvent(name='OperationCompletion', conformance=optional(choice=None))

Cluster RVC Operational State
	Accepted Commands changed:
		Stop
			XmlCommand(id=1, name='Stop', conformance=deprecated(choice=None))
			XmlCommand(id=1, name='Stop', conformance=disallowed(choice=None))
		Start
			XmlCommand(id=2, name='Start', conformance=deprecated(choice=None))
			XmlCommand(id=2, name='Start', conformance=disallowed(choice=None))
		GoHome
			XmlCommand(id=128, name='GoHome', conformance=otherwise(choice=None))
			XmlCommand(id=128, name='GoHome', conformance=optional(choice=None))
	Events changed:
		OperationCompletion
			XmlEvent(name='OperationCompletion', conformance=otherwise(choice=None))
			XmlEvent(name='OperationCompletion', conformance=optional(choice=None))

Cluster HEPA Filter Monitoring
	Features changed:
		ReplacementProductList (previously Replacement)
			XmlFeature(code='REP', name='Replacement', conformance=optional(choice=None))
			XmlFeature(code='REP', name='ReplacementProductList', conformance=optional(choice=None))

Cluster Activated Carbon Filter Monitoring
	Features changed:
		ReplacementProductList (previously Replacement)
			XmlFeature(code='REP', name='Replacement', conformance=optional(choice=None))
			XmlFeature(code='REP', name='ReplacementProductList', conformance=optional(choice=None))

Cluster Device Energy Management
	Revision change - old: 3 new: 4
	Features changed:
		ForecastAdjustment
			XmlFeature(code='FA', name='ForecastAdjustment', conformance=otherwise(choice=None))
			XmlFeature(code='FA', name='ForecastAdjustment', conformance=optional(choice=None))
		PowerAdjustment
			XmlFeature(code='PA', name='PowerAdjustment', conformance=optional(choice=Choice(marker='a', more=True)))
			XmlFeature(code='PA', name='PowerAdjustment', conformance=optional(choice=None))
		PowerForecastReporting
			XmlFeature(code='PFR', name='PowerForecastReporting', conformance=otherwise(choice=None))
			XmlFeature(code='PFR', name='PowerForecastReporting', conformance=otherwise(choice=None))
		ConstraintBasedAdjustment
			XmlFeature(code='CON', name='ConstraintBasedAdjustment', conformance=otherwise(choice=None))
			XmlFeature(code='CON', name='ConstraintBasedAdjustment', conformance=optional(choice=None))
		StateForecastReporting
			XmlFeature(code='SFR', name='StateForecastReporting', conformance=otherwise(choice=None))
			XmlFeature(code='SFR', name='StateForecastReporting', conformance=optional_wrapper(choice=Choice(marker='a', more=False)))
		StartTimeAdjustment
			XmlFeature(code='STA', name='StartTimeAdjustment', conformance=optional(choice=Choice(marker='a', more=True)))
			XmlFeature(code='STA', name='StartTimeAdjustment', conformance=optional(choice=None))
		Pausable
			XmlFeature(code='PAU', name='Pausable', conformance=optional(choice=Choice(marker='a', more=True)))
			XmlFeature(code='PAU', name='Pausable', conformance=optional(choice=None))
	Attributes changed:
		PowerAdjustmentCapability
			PowerAdjustmentCapability: datatype: list conformance: PA, access = R V
			PowerAdjustmentCapability: datatype: PowerAdjustCapabilityStruct conformance: PA, access = R V

Cluster Energy EVSE
	Revision change - old: 2 new: 3
	Features changed:
		ChargingPreferences
			XmlFeature(code='PREF', name='ChargingPreferences', conformance=provisional(choice=None))
			XmlFeature(code='PREF', name='ChargingPreferences', conformance=otherwise(choice=None))
		SoCReporting
			XmlFeature(code='SOC', name='SoCReporting', conformance=provisional(choice=None))
			XmlFeature(code='SOC', name='SoCReporting', conformance=otherwise(choice=None))
		PlugAndCharge
			XmlFeature(code='PNC', name='PlugAndCharge', conformance=provisional(choice=None))
			XmlFeature(code='PNC', name='PlugAndCharge', conformance=otherwise(choice=None))
		V2X
			XmlFeature(code='V2X', name='V2X', conformance=provisional(choice=None))
			XmlFeature(code='V2X', name='V2X', conformance=otherwise(choice=None))

Cluster Energy EVSE Mode
	Revision change - old: 1 new: 2
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: O, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: DEPONOFF, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Device Energy Management Mode
	Revision change - old: 1 new: 2
	Features changed:
		OnOff
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=optional(choice=None))
			XmlFeature(code='DEPONOFF', name='OnOff', conformance=disallowed(choice=None))
	Attributes changed:
		StartUpMode
			StartUpMode: datatype: uint8 conformance: O, access = RW VO
			StartUpMode: datatype: uint8 conformance: X, access = RW VO
		OnMode
			OnMode: datatype: uint8 conformance: DEPONOFF, access = RW VO
			OnMode: datatype: uint8 conformance: X, access = RW VO

Cluster Color Control
	Revision change - old: 6 new: 7
	Features changed:
		HueSaturation (previously Hue/Saturation)
			XmlFeature(code='HS', name='Hue/Saturation', conformance=optional(choice=None))
			XmlFeature(code='HS', name='HueSaturation', conformance=otherwise(choice=None))
		EnhancedHue (previously Enhanced)
			XmlFeature(code='EHUE', name='Enhanced', conformance=optional(choice=None))
			XmlFeature(code='EHUE', name='EnhancedHue', conformance=otherwise(choice=None))
		ColorLoop (previously Color)
			XmlFeature(code='CL', name='Color', conformance=optional(choice=None))
			XmlFeature(code='CL', name='ColorLoop', conformance=optional(choice=None))
		ColorTemperature (previously Color)
			XmlFeature(code='CT', name='Color', conformance=optional(choice=None))
			XmlFeature(code='CT', name='ColorTemperature', conformance=optional(choice=None))
	Attributes changed:
		DriftCompensation
			DriftCompensation: datatype: enum8 conformance: O, access = R V
			DriftCompensation: datatype: DriftCompensationEnum conformance: O, access = R V
		ColorMode
			ColorMode: datatype: enum8 conformance: M, access = R V
			ColorMode: datatype: ColorModeEnum conformance: M, access = R V
		EnhancedColorMode
			EnhancedColorMode: datatype: enum8 conformance: M, access = R V
			EnhancedColorMode: datatype: EnhancedColorModeEnum conformance: M, access = R V
		Options
			Options: datatype: map8 conformance: M, access = RW VO
			Options: datatype: OptionsBitmap conformance: M, access = RW VO
		ColorCapabilities
			ColorCapabilities: datatype: map16 conformance: M, access = R V
			ColorCapabilities: datatype: ColorCapabilitiesBitmap conformance: M, access = R V

Cluster Door Lock
	Revision change - old: 7 new: 8
	Features added: ['AliroProvisioning', 'AliroBLEUWB']
	Features removed: ['Logging', 'Notification']
	Features changed:
		User
			XmlFeature(code='USR', name='User', conformance=optional_wrapper(choice=None))
			XmlFeature(code='USR', name='User', conformance=otherwise(choice=None))
	Attributes added: ['AliroReaderVerificationKey', 'AliroReaderGroupIdentifier', 'AliroReaderGroupSubIdentifier', 'AliroExpeditedTransactionSupportedProtocolVersions', 'AliroGroupResolvingKey', 'AliroSupportedBLEUWBProtocolVersions', 'AliroBLEAdvertisingVersion', 'NumberOfAliroCredentialIssuerKeysSupported', 'NumberOfAliroEndpointKeysSupported']
	Attributes removed: ['NumberOfLogRecordsSupported', 'EnableLogging', 'KeypadOperationEventMask', 'RemoteOperationEventMask', 'ManualOperationEventMask', 'RFIDOperationEventMask', 'KeypadProgrammingEventMask', 'RemoteProgrammingEventMask', 'RFIDProgrammingEventMask']
	Attributes changed:
		SecurityLevel
			SecurityLevel: datatype: UNKNOWN conformance: D, access = RW N/AN/A
			SecurityLevel: datatype: UNKNOWN conformance: D, access = R V
	Accepted Commands added: ['SetAliroReaderConfig', 'ClearAliroReaderConfig']
	Accepted Commands removed: ['GetLogRecord']
	Generated Commands removed: ['GetLogRecordResponse']

Cluster Window Covering
	Attributes changed:
		Mode
			Mode: datatype: ModeBitmap conformance: M, access = RW VM
			Mode: datatype: ModeBitmap Type. conformance: M, access = RW VM

Cluster Application Launcher
	Revision change - old: 1 new: 2

Device Types newly added in 1.4
['Heat Pump', 'Mounted Dimmable Load Control', 'Network Infrastructure Manager', 'Thread Border Router', 'Solar Power', 'Joint Fabric Administrator', 'Mounted On/Off Control', 'Water Heater', 'Battery Storage', 'Secondary Network Interface']

Device Types removed since 1.3
[]

Device Type Window Covering
	Server Clusters removed: ['Scenes Management']

Device Type Pump Controller
	Revision change - old: 3 new: 4
	Client Clusters removed: ['Binding']

Device Type Window Covering Controller
	Client Clusters removed: ['Scenes Management']

Device Type Door Lock
	Server Clusters added: ['Scenes Management']
	Server Clusters removed: ['Scenes Management']

Device Type Energy EVSE
	Revision change - old: 1 new: 2

Device Type Aggregator
	Revision change - old: 1 new: 2
	Server Clusters added: ['Commissioner Control']

Device Type Door Lock Controller
	Client Clusters added: ['Scenes Management']
	Client Clusters removed: ['Scenes Management']

Device Type Bridged Node
	Revision change - old: 2 new: 3
	Server Clusters added: ['Administrator Commissioning', 'Ecosystem Information']

Device Type Device Energy Management
	Revision change - old: 1 new: 2
	Server Clusters changed:
		Device Energy Management
			Device Energy Management: P, M
			Device Energy Management: M
		Device Energy Management Mode
			Device Energy Management Mode: P, M
			Device Energy Management Mode: ControllableESA, O

Device Type Root Node
	Revision change - old: 2 new: 3
	Server Clusters changed:
		ICD Management
			ICD Management: SIT
			ICD Management: SIT | LIT

Device Type Thermostat
	Revision change - old: 3 new: 4
	Server Clusters removed: ['Alarms', 'Time Synchronization', 'Scenes Management']
	Server Clusters changed:
		Energy Preference
			Energy Preference: P, O
			Energy Preference: O
	Client Clusters removed: ['Time', 'Time Synchronization']

Device Type Fan
	Revision change - old: 2 new: 3
	Server Clusters added: ['On/Off']

Device Type Air Purifier
	Revision change - old: 1 new: 2
	Server Clusters added: ['On/Off']

Device Type Occupancy Sensor
	Revision change - old: 3 new: 4
	Server Clusters added: ['Boolean State Configuration']

Device Type Refrigerator
	Revision change - old: 1 new: 2

Device Type Temperature Controlled Cabinet
	Revision change - old: 2 new: 3

Device Type Robotic Vacuum Cleaner
	Revision change - old: 2 new: 3
	Server Clusters added: ['Service Area']

Device Type Oven
	Revision change - old: 1 new: 2

Provisional elements:
=================
Provisional Clusters
	['Ballast Configuration', 'Content Control', 'Energy Preference', 'Joint Fabric Datastore Cluster', 'Joint Fabric PKI', 'ProxyConfiguration', 'ProxyDiscovery', 'Scenes Management', 'ValidProxies']

Door Lock
	Provisional features: ['FingerCredentials', 'FaceCredentials']

GroupKeyManagement
	Provisional features: ['CacheAndSync']

General Commissioning
	Provisional features: ['TermsAndConditions']
	Provisional attributes: ['TCAcceptedVersion', 'TCMinRequiredVersion', 'TCAcknowledgements', 'TCAcknowledgementsRequired', 'TCUpdateDeadline']
	Provisional accepted commands: ['SetTCAcknowledgements']
	Provisional generated commands: ['SetTCAcknowledgementsResponse']

Energy EVSE
	Provisional features: ['ChargingPreferences', 'SoCReporting', 'PlugAndCharge', 'V2X']

Thread Network Diagnostics
	Provisional attributes: ['ExtAddress', 'Rloc16']

Level Control
	Provisional features: ['Frequency']

Microwave Oven Control
	Provisional features: ['PowerInWatts']
	Provisional attributes: ['SupportedWatts', 'SelectedWattIndex']

Dimmable Light
	Provisional server clusters: ['Scenes Management']

Color Dimmer Switch
	Provisional client clusters: ['Scenes Management']

On/Off Light Switch
	Provisional client clusters: ['Scenes Management']

Color Temperature Light
	Provisional server clusters: ['Scenes Management']

Mounted Dimmable Load Control
	Provisional server clusters: ['Scenes Management']

Dimmer Switch
	Provisional client clusters: ['Scenes Management']

Control Bridge
	Provisional client clusters: ['Scenes Management']

Casting Video Client
	Provisional client clusters: ['Content Control']

On/Off Plug-in Unit
	Provisional server clusters: ['Scenes Management']

On/Off Sensor
	Provisional client clusters: ['Scenes Management']

Pump Controller
	Provisional client clusters: ['Scenes Management']

Room Air Conditioner
	Provisional server clusters: ['Scenes Management']

Video Remote Control
	Provisional client clusters: ['Content Control']

Pump
	Provisional server clusters: ['Scenes Management']

On/Off Light
	Provisional server clusters: ['Scenes Management']

Door Lock Controller
	Provisional client clusters: ['Scenes Management']

Basic Video Player
	Provisional server clusters: ['Content Control']

Mounted On/Off Control
	Provisional server clusters: ['Scenes Management']

Casting Video Player
	Provisional server clusters: ['Content Control']

Extended Color Light
	Provisional server clusters: ['Scenes Management']

Dimmable Plug-In Unit
	Provisional server clusters: ['Scenes Management']

### CODE REVIEW

**Review by robszewczyk — APPROVED** [review]

**Comment by jamesharrow** [comment]
EVSE PREF should have had P removed.

**Review by jamesharrow — CHANGES_REQUESTED** [review]
EVSE PREF should have had P removed (see new spec PR https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10597)

File: data_model/1.4/device_types/RootNodeDeviceType.xml [hunk 1]
```diff
@@ -160,16 +160,16 @@ Davis, CA 95616, USA
...
       </mandatoryConform>
       <features>
         <feature code="" name="LongIdleTimeSupport">
-          <otherwiseConform>
-            <provisionalConform/>
-            <mandatoryConform>
-              <feature name="LIT"/>
-            </mandatoryConform>
-          </otherwiseConform>
+          <mandatoryConform>
```
Comment (cjandhyala) [inline data_model/1.4/device_types/RootNodeDeviceType.xml:170]: is this mandatory or optional ?

**Review by jamesharrow — APPROVED** [review]

File: data_model/1.4/clusters/EnergyEVSE.xml [hunk 2]
```diff
@@ -240,7 +240,6 @@ Davis, CA 95616, USA
     <struct name="ChargingTargetScheduleStruct">
```
Comment (jamesharrow) [inline data_model/1.4/clusters/EnergyEVSE.xml:240]: EVSE PREF should not be P

File: data_model/1.4/device_types/RootNodeDeviceType.xml (hunk 1 above)
Comment (cecille) [reply data_model/1.4/device_types/RootNodeDeviceType.xml:170]: mandatory. This is correct, matches master.

File: data_model/1.4/clusters/EnergyEVSE.xml (hunk 2 above)
Comment (cecille) [reply data_model/1.4/clusters/EnergyEVSE.xml:237]: Resolved in the latest spec pull.

**Review by ReneJosefsen — APPROVED** [review]

**Pull request merged**